lib/puppet/defaults.rb

```diff
@@ -1470,6 +1470,12 @@ def self.default_vendormoduledir
       :default    => lambda { Puppet::Settings.domain_fact },
       :desc       => "The domain which will be queried to find the SRV records of servers to use.",
     },
+    :http_extra_headers => {
+      :default    => [],
+      :type       => :http_extra_headers,
+      :desc       => "The list of extra headers that will be sent with every HTTP request.
```

Comment:
This doesn't seem to be happening with _every_ request as noted here, just for downloading file content. Was this intentional?